api/src/main/java/com/spotify/spydra/api/process/ProcessHelper.java

```diff
@@ -45,22 +46,29 @@ public static int executeCommand(List<String> command) throws IOException {
     return p.exitValue();
   }
 
-  public static String executeForOutput(List<String> command) throws IOException {
+  public static boolean executeForOutput(List<String> command, StringBuilder outputBuilder)
+      throws IOException {
     LOGGER.debug("Executing command: " + StringUtils.join(command, " "));
     ProcessBuilder pb = new ProcessBuilder(command)
-        .redirectError(ProcessBuilder.Redirect.INHERIT)
+        .redirectError(ProcessBuilder.Redirect.PIPE)
         .redirectOutput(ProcessBuilder.Redirect.PIPE);
 
     Process p = pb.start();
     try {
-      BufferedReader in = new BufferedReader(new InputStreamReader(p.getInputStream()));
-      StringBuilder builder = new StringBuilder();
+      int exitCode = p.waitFor();
+      boolean success = exitCode == Shell.SUCCESS;
+      BufferedReader in;
+      if (success) {
```

Comment:
Could be the case where both stdout and stderr have contents to display? In that case we will be ignoring one...Do you think that could be a problem?

Reply:
Not really. As the error logging is more of a convenience here in case of errors, and it was complete for couple of error scenarios I tested (invalid parameters, no valid credentials). For success case the behaviour won't change, as for success values JSON is expected.